internal/internal_event_handlers.go

```diff
@@ -1077,45 +1075,69 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 	eventID int64,
 	attributes *eventpb.MarkerRecordedEventAttributes,
 ) error {
-	switch attributes.GetMarkerName() {
-	case sideEffectMarkerName:
-		for sideEffectID, result := range attributes.GetDetails() {
-			weh.sideEffectResult[sideEffectID] = result
-		}
-		return nil
-	case versionMarkerName:
-		for changeID, versionPayload := range attributes.GetDetails() {
-			var version Version
-			_ = weh.dataConverter.FromData(versionPayload, &version)
-			weh.changeVersions[changeID] = version
-			weh.decisionsHelper.handleVersionMarker(eventID, changeID)
-		}
-		return nil
-	case localActivityMarkerName:
-		err := weh.handleLocalActivityMarker(attributes.GetDetails(), attributes.GetFailure())
-		if err != nil {
-			return fmt.Errorf("marker name %q: %w", localActivityMarkerName, err)
-		}
-		return nil
-	case mutableSideEffectMarkerName:
-		for fixedID, result := range attributes.GetDetails() {
-			weh.mutableSideEffect[fixedID] = result
+	var err error
+	if attributes.GetDetails() == nil {
+		err = ErrMissingMarkerDetails
+	} else {
+		switch attributes.GetMarkerName() {
+		case sideEffectMarkerName:
+			if sideEffectIDPayload, ok := attributes.GetDetails()[sideEffectMarkerIDName]; !ok {
```

Comment:
consider moving all the cases into separate functions the same way handleLocalActivityMarker is already done.

Reply:
I had this idea too but decided to leave it as is for now.